### ISSUE TITLE
Controlling the output terminal colors

### ISSUE THREAD

**Issue by chriscroome** [ansible/ansible-lint#4469]
I have a terminal set to white on black, `ansible-lint` warnings are black on black, I have this in `ansible.cfg` to fix an issue like this with Ansible itself:

```ini
[colors]
debug = blue
```

But I can't work out how to set the `ansible-lint` warning colour, can someone explain how this can be done?

_Originally posted by @chriscroome in https://github.com/ansible/ansible-lint/discussions/4461_

_Discussed on the Ansible Forum in https://forum.ansible.com/t/how-can-the-ansible-lint-warning-colour-be-controlled/39511_

**Comment by chriscroome** [comment]
Sorry I don't know what Rich is but I have only recently noticed this issue, I haven't tested with older versions to see if these have the issue but could, do you know which version should I roll back to for testing?

**Comment by MindTooth** [comment]
https://github.com/ansible/ansible-lint/pull/4396

They refactored out [Rich](https://github.com/Textualize/rich) which I assume provides the colors, etc.  So now that it's out, we have black on black.  Which for me is a bug/regression. 😔  I hope that explained it.

`ansible-lint==24.10.0` works for me and now prints gray which is visible. _I have only code from work and can't show it._

@ssbarnea I see you made the pull request.  Thoughts?

**Comment by osiris** [comment]
Hi @ssbarnea in ansible-lint==v24.12.2 the black on black error persists!

![Image](https://github.com/user-attachments/assets/93f43360-8b3e-448e-a01e-8d6387783c6f)

**Comment by aphedges** [comment]
For anyone that needs a workaround, @chriscroome noted in the original forum thread (<https://forum.ansible.com/t/how-can-the-ansible-lint-warning-colour-be-controlled/39511/3>) that one can disable the unreadable colors with `--nocolor`. That change was enough to allow me to upgrade.

**Comment by ssbarnea** [comment]
Yes, removal of rich does change the way we produce colors but was not supposed to produce such undesirable outcome. 99% of users as using white on black anyway, including myself. I do suspect that your terminal might be missing some features that we use and that is why we see this problem.

Can you please try several terminals in order detect why you get this behavior?

Note that we do not have a plan to implement any feature to allow customization of colors (at least yet), but we will try to fix the output to make it readable. We only need a way to reproduce it.  Maybe running an ANSI test with your terminal might uncover which ANSI extensions are not implemented by it?

**Comment by MindTooth** [comment]
Here is from both Terminal.app and kitty:

![Image](https://github.com/user-attachments/assets/e5f80a4e-142a-4418-97bd-14a975bb9c6b)
![Image](https://github.com/user-attachments/assets/7bc95722-99c3-44b1-971b-c117304ffad8)

In contrast to 24.10.0:

![Image](https://github.com/user-attachments/assets/280a81e2-5c87-467d-ad96-cf0b22e966c4)

**Comment by chriscroome** [comment]
I'm also using Kitty, which is, of course, [installed using Ansible](https://git.coop/webarch/kitty)! 😉

**Comment by aphedges** [comment]
> Can you please try several terminals in order detect why you get this behavior?

I'm using macOS, and I get the unreadable text in PyCharm 2024.3.1.1 (top) but not in iTerm2 3.5.11 (bottom). I've encountered various other problems with colors in PyCharm's terminal, so I wouldn't be surprised if there is a bug in the terminal itself, not `ansible-lint`'s code.

![Screenshot of `ansible-lint` output in PyCharm's terminal](https://github.com/user-attachments/assets/74ec4798-a839-45db-8fa8-89014bd7e310)

![Screenshot of `ansible-lint` output in iTerm2](https://github.com/user-attachments/assets/18305c46-f601-4a65-bc8c-c124aae55dc3)

> Maybe running an ANSI test with your terminal might uncover which ANSI extensions are not implemented by it?

Do you have recommendations for one to run? That seems like a useful tool to help with terminal problems like these.

**Comment by chriscroome** [comment]
With the Debian Bookworm Gnome terminal I get black-on-black and unlike with Kitty the text doesn't even show up when highlighted:

```bash
gnome-terminal --version
# GNOME Terminal 3.46.8 using VTE 0.70.6 +BIDI +GNUTLS +ICU +SYSTEMD
``` 

With Foot I get just-about-readable grey on grey:

```bash
foot --version
foot version: 1.13.1 +pgo +ime +graphemes -assertions
```

So perhaps it is a client issue? 🤷

**Comment by chriscroome** [comment]
Thanks @ssbarnea I have [started a discussion](https://github.com/kovidgoyal/kitty/discussions/8241) about this on the Kitty forum.

**Comment by chriscroome** [comment]
The [developer of Kitty is of the view that](https://github.com/kovidgoyal/kitty/discussions/8241#discussioncomment-11902297):

> I didnt look at the dump file but this is obviously a client issue. The
> terminal emulator simply uses whatever colors the client tells it too.
> One thing the client should make sure of is that it is not using bold to
> get bright colors, instead it needs to use colors from 8-15 for the
> right versions of colors from 0-7.

**Comment by MindTooth** [comment]
Since I'm using [Srcery](https://srcery.sh) and based on the comment made from Kitty, I did check the scheme.  What I found is that the color black is the same as the background, so black-on-black.

From https://github.com/kovidgoyal/kitty-themes/, I can see a lot of the original themes too have black-on-black, which could explain this phenomena.

I will look more into this at least for Srcery, and it's something to have in mind for other schemes as well.

**Comment by aphedges** [comment]
Given that this bug appears on 3 out of 4 terminals tested, I feel the fix should probably be made in `ansible-lint` itself.

I'm not saying we should go back to using `rich`, but `rich` was clearly doing something that avoided this problem.

**Comment by osiris** [comment]
Hi @ssbarnea, I confirm that the _black on black error_ was solved in `ansible-lint==v25.1.1`
Tested at `Terminator 2.1.2` in `Debian 12`

![Image](https://github.com/user-attachments/assets/123ff478-ac73-4141-b2d8-0ac21553db72)
